src/lib/sticky-header/sticky-header.spec.ts

```diff
@@ -0,0 +1,180 @@
+import {async, ComponentFixture, TestBed, fakeAsync, tick} from '@angular/core/testing';
+import {Component, DebugElement, ViewChild} from '@angular/core';
+import {StickyHeaderModule, CdkStickyRegion, CdkStickyHeader,
+  STICKY_HEADER_SUPPORT_STRATEGY} from './index';
+import {OverlayModule, Scrollable} from '../core/overlay/index';
+import {PlatformModule} from '../core/platform/index';
+import {By} from '@angular/platform-browser';
+import {dispatchFakeEvent} from '@angular/cdk/testing';
+
+
+
+describe('sticky-header with positioning not supported', () => {
+  let fixture: ComponentFixture<StickyHeaderTest>;
+  let testComponent: StickyHeaderTest;
+  let stickyElement: DebugElement;
+  let stickyParentElement: DebugElement;
+  let scrollableElement: DebugElement;
+  let stickyHeader: CdkStickyHeader;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [ OverlayModule, PlatformModule, StickyHeaderModule ],
+      declarations: [StickyHeaderTest],
+      providers: [
+        {provide: STICKY_HEADER_SUPPORT_STRATEGY, useValue: false},
+      ],
+    });
+    TestBed.compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(StickyHeaderTest);
+    fixture.detectChanges();
+    testComponent = fixture.debugElement.componentInstance;
+    stickyElement = fixture.debugElement.query(By.directive(CdkStickyHeader));
+    stickyParentElement = fixture.debugElement.query(By.directive(CdkStickyRegion));
+    stickyHeader = stickyElement.injector.get<CdkStickyHeader>(CdkStickyHeader);
+    scrollableElement = fixture.debugElement.query(By.directive(Scrollable));
+  });
+
+  it('should be able to find stickyParent', () => {
+    expect(stickyHeader.stickyParent).not.toBe(null);
+  });
+
+  it('should be able to find scrollableContainer', () => {
+    expect(stickyHeader.upperScrollableContainer).not.toBe(null);
+  });
+
+  it('should stick in the right place when scrolled to the top of the container', fakeAsync(() => {
+    let scrollableContainerTop = stickyHeader.upperScrollableContainer
+      .getBoundingClientRect().top;
+    expect(stickyHeader.element.getBoundingClientRect().top).not.toBe(scrollableContainerTop);
+    tick();
+
+    // Scroll the scrollableContainer up to stick
+    fixture.componentInstance.scrollDown();
+    // Use tick(100) to tick forward to let stickyHeader's _applyStickyPositionStyles()
+    // function finished
```

Comment:
Because after we calling `fixture.componentInstance.scrollDown();`, the s`crollDown() `function dispatches a fake scroll event, which causes the `_applyStickyPositionStyles` being called. And When it finds that the header needs to be stuck, it will reassign value to the header , so the header will be repainted. And we need to wait till animation has finished.

Changed the comments to `//wait till animation has finished'`